### PR TITLE
Adds `fromHeaders` and `fromParams` fields for RequestAuthentication

### PR DESCRIPTION
**Pull request by videlov**
**Description**

Changes proposed in this pull request:

- Adds `fromHeaders` and `fromParams` fields for RequestAuthentication

**Related issue(s)**

https://github.com/kyma-project/api-gateway/issues/279

### CODE REVIEW

File: api/v1beta1/apirule_types.go [hunk 1]
```diff
@@ -165,8 +165,19 @@ type JwtAuthorization struct {
...
 	Issuer  string `json:"issuer"`
 	JwksUri string `json:"jwksUri"`
+	// +optional
+	FromHeaders []*JwtHeader `json:"fromHeaders,omitempty"`
+	// +optional
+	FromParams []string `json:"fromParams,omitempty"`
+}
+
+// JwtHeader for specifying from header for the Jwt token
+type JwtHeader struct {
```
Comment (barchw) [inline api/v1beta1/apirule_types.go:179]: Please run `make generate manifests`

**Review by barchw — CHANGES_REQUESTED** [review]

**Review by barchw — CHANGES_REQUESTED** [review]

File: internal/processing/istio/jwt_validator.go [hunk 2]
```diff
@@ -118,3 +207,58 @@ func containsSidecar(pod corev1.Pod) bool {
...
+					failures = append(failures, validation.Failure{AttributePath: attributePath, Message: "Can't read json: " + err.Error()})
+					return failures
+				}
+
+				for k, authentication := range template.Authentications {
+					jwtAuthKey := authentication.Issuer + authentication.JwksUri
+					if jwtAuths[jwtAuthKey] != nil && !isJwtAuthenticationsEqual(authentication, jwtAuths[jwtAuthKey]) {
+						attributeSubPath := fmt.Sprintf("%s%s[%d]", attributePath, ".config.authentications", k)
+						failures = append(failures, validation.Failure{AttributePath: attributeSubPath, Message: "multiple jwt configurations that differ for the same issuer"})
+					} else {
```
[the remark this answers is not in the document]
Comment (barchw) [reply internal/processing/istio/jwt_validator.go:233]: Oh, it is taking the Issuer AND JWKS Uri. Sorry for the confusion :)

File: tests/integration/features/istio-jwt/istio_jwt.feature [hunk 3]
```diff
@@ -87,4 +86,18 @@ Feature: Exposing endpoints with Istio JWT authorization strategy
...
+    Given JwtTokenFromHeaders: There is a httpbin service
+    When JwtTokenFromHeaders: The APIRule is applied
+    Then JwtTokenFromHeaders: Calling the "/headers" endpoint without a token should result in status between 400 and 403
+    And JwtTokenFromHeaders: Calling the "/headers" endpoint with a valid "JWT" token from header "x-jwt-token" and prefix "JwtToken" should result in status between 200 and 299
+    And JwtTokenFromHeaders: Teardown httpbin service
+
+  Scenario: Calling a httpbin endpoint secured with different JWT token from params
+    Given JwtTokenFromParams: There is a httpbin service
+    When JwtTokenFromParams: The APIRule is applied
+    And JwtTokenFromParams: Calling the "/ip" endpoint without a token should result in status between 400 and 403
```
Comment (barchw) [inline tests/integration/features/istio-jwt/istio_jwt.feature:101]: Could you add a negative check for token in default `authorization` header?

File: tests/integration/features/istio-jwt/istio_jwt.feature [hunk 4]
```diff
@@ -87,4 +86,18 @@ Feature: Exposing endpoints with Istio JWT authorization strategy
...
     When JwtIssuerJwksNotMatch: The APIRule is applied
     And JwtIssuerJwksNotMatch: Calling the "/ip" endpoint with a valid "JWT" token should result in body containing "Jwks doesn't have key to match kid or alg from Jwt"
-    And JwtIssuerJwksNotMatch: Teardown httpbin service
\ No newline at end of file
+    And JwtIssuerJwksNotMatch: Teardown httpbin service
+
+  Scenario: Calling a httpbin endpoint secured with different JWT token from headers
+    Given JwtTokenFromHeaders: There is a httpbin service
+    When JwtTokenFromHeaders: The APIRule is applied
+    Then JwtTokenFromHeaders: Calling the "/headers" endpoint without a token should result in status between 400 and 403
```
Comment (barchw) [inline tests/integration/features/istio-jwt/istio_jwt.feature:94]: Could you add a negative check for token in default `authorization` header?

**Review by barchw — CHANGES_REQUESTED** [review]

File: tests/integration/features/istio-jwt/istio_jwt.feature (hunk 4 above)
Comment (videlov) [reply tests/integration/features/istio-jwt/istio_jwt.feature:94]: Yes this makes sense, good catch. Incoming.

File: tests/integration/features/istio-jwt/istio_jwt.feature (hunk 3 above)
Comment (videlov) [reply tests/integration/features/istio-jwt/istio_jwt.feature:101]: Yes this makes sense, good catch. Incoming.

**Review by barchw — APPROVED** [review]
/meow

**Pull request merged**